modules/backend/assets/js/october.tabformexpandcontrols.js

```diff
@@ -160,4 +160,84 @@
     }
 
     $.oc.tabFormExpandControls = TabFormExpandControls
-}(window.jQuery);
\ No newline at end of file
+
+}(window.jQuery);
+
+/*
+ * Auto update WAI-ARIA when a user updates the tabs
+ */
+ $(document).ready(function() {
+
+     /* If no selected => select first */
+     if ($('.master-tabs a[aria-selected]').length === 0) {
+         $('.master-tabs a:first').attr({
+             'aria-selected': 'true',
+             'tabindex': '0'
+         });
+     } else if ($('.primary-tabs a[aria-selected]').length === 0) {
+         $('.primary-tabs a:first').attr({
+             'aria-selected': 'true',
+             'tabindex': '0'
+         });
+     } else if ($('.secondary-tabs a[aria-selected]').length === 0) {
+         $('.secondary-tabs a:first').attr({
+             'aria-selected': 'true',
+             'tabindex': '0'
+         });
+     } else if ($('.content-tabs a[aria-selected]').length === 0) {
+         $('.content-tabs a:first').attr({
+             'aria-selected': 'true',
+             'tabindex': '0'
+         });
+     }
+
+     /* Update wai-aria on navigation */
+     $('body').on('click keydown', '.master-tabs a,.primary-tabs a,.secondary-tabs a,.content-tabs a', function(event) {
+
```

Comment:
Filter out keydown events:
```
var whitelist = [
    'ArrowLeft',
    'ArrowUp',
    'ArrowDown',
    'ArrowRight',
    'Home',
    'End'
];
if(event.type === 'keydown' && !whitelist.includes(event.key)){
    return;
}```

Reply:
You will need to whitelist a few more keys. `Tab`, `Space/Space Bar` and `Enter/Return`. The browser api is using them.

For example, I press the `Tab` key on each tab the WAI-ARIA will update and change on these code lines:

https://github.com/octobercms/october/blob/0d18e1b8e4907f8d9c21e888a4c88c6c00dbb615/modules/backend/assets/js/october.tabformexpandcontrols.js#L210-L214

So I think it will be like this:

```js
var whitelist = [
    'ArrowLeft',
    'ArrowUp',
    'ArrowDown',
    'ArrowRight',
    'Home',
    'End',
    'Tab',
    'Enter',
    '(Space character)',
    'Spacebar',
    ' '
];
if(event.type === 'keydown' && !whitelist.includes(event.key)){
    return;
}
```

- Space bar is a bit of a headache, I did some research on it 2 days ago and came to the conclusion you need three fields. This is what I found out:

Future propsoal - `(Space character)`
Current proposal - ` `
Passed propsal (older browsers e.g. IE11) - `Spacebar`

Reply:
Yeah-yeah, @ayumihamsaki was an example made from a mobile phone. So, you're right 👍

Reply:
I'm just writing my commit description here 😄